tests/keithley_2700.py

```diff
@@ -0,0 +1,188 @@
+import unittest
+from unittest import skipIf
+from utils.channel_access import ChannelAccess
+from utils.testing import get_running_lewis_and_ioc
+from utils.ioc_launcher import IOCRegister
+
+
+
+class Keithley_2700Tests(unittest.TestCase):
+    """
+    Tests for the Keithley2700.
+    """
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700")
+        self.ca = ChannelAccess(default_timeout=30)
+        self.ca.wait_for("KHLY2700_01:IDN")
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_idn_defined_WHEN_read_THEN_idn_is_as_expected(self):
+        self.ca.assert_that_pv_is("KHLY2700_01:IDN", "Keithley 2700 Multimeter emulator.")
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_available_AND_reserved_bytes_set_WHEN_read_THEN_staTus_of_memory_is_as_expected(self):
+        self.ca.set_pv_value("KHLY2700_01:BUFFER_SIZE:SP", 100)
+        self._lewis.backdoor_set_on_device("buffer_readings", 10)
+        self._lewis.backdoor_set_on_device("buffer", 100)
+        self.ca.assert_that_pv_is("KHLY2700_01:BUFFER_SIZE.", 100)
+        self.ca.set_pv_value("KHLY2700_01:BUFFER_STATS.PROC", 1)
+        self.ca.assert_that_pv_is("KHLY2700_01:BUFFER_STATS", "92160 bytes,10240 bytes")
```

Comment:
There is a method in `ca` for simultaneously setting a PV and checking the readback matches the expected value. Please use that instead and apply elsewhere in the test file